g++ compilation errors

With `gcc version 11.3.0 (conda-forge gcc 11.3.0-19)`, building `stf_protocol_types.cpp.o` results in a few compilation errors:
```
stf_lib/stf-inc/protocols/tilelink.hpp:23:19: error: 'stf::protocols::tilelink::fields::Code' declared with greater visibility than its base 'stf::ProtocolField<stf::protocols::tilelink::fields::Code, unsigned char>' [-Werror=attributes]
   23 |             FIELD(Code, uint8_t);
      |                   ^~~~
stf_lib/stf-inc/stf_protocol_fields.hpp:39:11: note: in definition of macro '__FIELD'
   39 |     class field_name : public STF_UNPACK_TEMPLATE(parent_class_tuple) { \
      |           ^~~~~~~~~~
stf_lib/stf-inc/stf_protocol_fields.hpp:85:36: note: in expansion of macro '_PROTOCOL_FIELD'
   85 | #define _FIELD_1(field_name, type) _PROTOCOL_FIELD(field_name, type)
      |                                    ^~~~~~~~~~~~~~~
/miniconda3/envs/sparta/include/boost/preprocessor/cat.hpp:29:34: note: in expansion of macro '_FIELD_1'
   29 | #    define BOOST_PP_CAT_I(a, b) a ## b
      |                                  ^
stf_lib/stf-inc/protocols/tilelink.hpp:23:13: note: in expansion of macro 'FIELD'
   23 |             FIELD(Code, uint8_t);
```

--- a/stf-inc/stf_protocol_fields.hpp
+++ b/stf-inc/stf_protocol_fields.hpp
@@ -17,6 +17,14 @@
 #include "stf_serializable_container.hpp"
 #include "type_utils.hpp"
 
+// Conda adds -fvisibility-inlines-hidden to the default compiler flags, which gives a compiler error unless
+// we add this visibility tag to field class definitions
+#if __GNUC__ >= 4
+    #define DEFAULT_VIS __attribute__((visibility("default")))
+#else
+    #define DEFAULT_VIS
+#endif
+
 // Defines r-value and l-value field class constructors for the given argument type
 // parent_class_tuple and arg_type should either be bare type names or tuples packed with STF_PACK_TEMPLATE
 #define __FIELD_CONSTRUCTOR(parent_class_tuple, field_name, arg_type)                                   \
@@ -36,7 +44,7 @@
 
 // Declares a field class with the given parent type, name, and constructor argument types
 #define __FIELD(parent_class_tuple, field_name, ...)                    \
-    class field_name : public STF_UNPACK_TEMPLATE(parent_class_tuple) { \
+    class DEFAULT_VIS field_name : public STF_UNPACK_TEMPLATE(parent_class_tuple) { \
         public:                                                         \
             field_name() = default;                                     \
             explicit field_name(STFIFstream& reader) :                  \

